app/src/main/java/com/supertokens/session/SuperTokensHttpURLConnection.java

```diff
@@ -102,117 +129,121 @@ public static HttpURLConnection newRequest(URL url, PreConnectCallback preConnec
         try {
             while (true) {
                 HttpURLConnection connection;
-                String preRequestIdRefreshToken;
+                SuperTokensCustomHttpURLConnection customConnection;
+                Utils.LocalSessionState preRequestLocalSessionState;
                 int responseCode;
                 // TODO: write comment as to why we have this lock here. Do we also have this lock for iOS and website package?
                 refreshAPILock.readLock().lock();
                 try {
                     connection = (HttpURLConnection) url.openConnection();
+                    customConnection = new SuperTokensCustomHttpURLConnection(connection, applicationContext);
 
                     // Add antiCSRF token, if present in storage, to the request headers
-                    preRequestIdRefreshToken = IdRefreshToken.getToken(applicationContext);
-                    String antiCSRFToken = AntiCSRF.getToken(applicationContext, preRequestIdRefreshToken);
+                    preRequestLocalSessionState = Utils.getLocalSessionState(applicationContext);
+                    String antiCSRFToken = AntiCSRF.getToken(applicationContext, preRequestLocalSessionState.lastAccessTokenUpdate);
 
                     if (antiCSRFToken != null) {
-                        connection.setRequestProperty(Constants.CSRF_HEADER_KEY, antiCSRFToken);
+                        customConnection.setRequestProperty(Constants.CSRF_HEADER_KEY, antiCSRFToken);
                     }
 
                     // Get the default cookie manager that is used, if null set a new one
-                    if (CookieManager.getDefault() == null) {
+                    if (CookieManager.getDefault() == null && SuperTokens.config.tokenTransferMethod.equals("cookie")) {
```

Comment:
the setting for cookie can be set from the backend as well.. so we should detect if the backend is sending cookies (with sAccessToken) and if it is, then we should throw an error in case CookieManager is not initialised